tools/VscPreviewExtension/src/childProcessHost.ts

```diff
@@ -74,24 +65,62 @@ export class ChildProcessHost {
         let docContent = doc.getText();
         let fileName = doc.fileName;
         let rootPath = workspace.rootPath;
-        let filePath;
+        let relativePath;
         if (!rootPath || !fileName.includes(rootPath)) {
-            let indexOfFilename = fileName.lastIndexOf("\\");
-            rootPath = fileName.substr(0, indexOfFilename);
-            filePath = fileName.substring(indexOfFilename + 1);
+            let indexOfFileName = fileName.lastIndexOf("\\");
+            rootPath = fileName.substr(0, indexOfFileName);
+            relativePath = fileName.substring(indexOfFileName + 1);
         } else {
             let rootPathLength = rootPath.length;
-            filePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
+            relativePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
         }
-        if (doc.languageId === ConstVariable.languageId) {
-            let numOfRow = doc.lineCount;
-            this.writeToStdin(rootPath, filePath, numOfRow, docContent);
+        if (doc.languageId === "markdown") {
+            this.sendHttpRequestCoreAsync(rootPath, relativePath, docContent);
```

Comment:
>this.sendHttpRequestCoreAsync(rootPath, relativePath, docContent); [](start = 12, length = 66)

do we have to await this async function?@Qinen.
In my opinion, we just need to make a request and call the callback function when the result generated. Use await will get the same result. Your reason is the debuggable property when some error happened, we need to do some investigation